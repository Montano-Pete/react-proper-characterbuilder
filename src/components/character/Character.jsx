import React from 'react';

export default function Character({ head, torso, pants }) {
  return (
    <section className="entire-character">
      <div className="head">
        <img
          aria-label="head"
          alt="character-head"
          src={`${head}-head.png`}
          width="200"
          height="200"
        ></img>
      </div>

      <div className="torso">
        <img
          aria-label="torso"
          alt="character-torso"
          src={`${torso}-torso.png`}
          width="200"
          height="200"
        ></img>
      </div>

      <div className="pant">
        <img
          aria-label="pant"
          alt="character-pants"
          src={`${pants}-pants.png`}
          width="200"
          height="200"
        ></img>
      </div>
    </section>
  );
}
