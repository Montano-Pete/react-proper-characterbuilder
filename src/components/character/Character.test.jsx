import { render, screen } from '@testing-library/react';
import Character from './Character';

it('should render Controls elements on page', () => {
  const { container } = render(
    <Character head="duck" torso="pink" pant="white" />
  );

  const head = screen.getByLabelText(/head/);
  const torso = screen.getByLabelText(/torso/);
  const pant = screen.getByLabelText(/pant/);

  expect(head).toBeInTheDocument();
  expect(torso).toBeInTheDocument();
  expect(pant).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});
