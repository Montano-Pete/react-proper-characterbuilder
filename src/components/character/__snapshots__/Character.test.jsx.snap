// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Controls elements on page 1`] = `
<div>
  <section
    class="entire-character"
  >
    <div
      class="head"
    >
      <img
        alt="character-head"
        aria-label="head"
        height="200"
        src="duck-head.png"
        width="200"
      />
    </div>
    <div
      class="torso"
    >
      <img
        alt="character-torso"
        aria-label="torso"
        height="200"
        src="pink-torso.png"
        width="200"
      />
    </div>
    <div
      class="pant"
    >
      <img
        alt="character-pants"
        aria-label="pant"
        height="200"
        src="undefined-pants.png"
        width="200"
      />
    </div>
  </section>
</div>
`;
